utils/date.go

```diff
@@ -0,0 +1,123 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+//
+// Calculations involving schedule hours and start-of-day.
+//
+
+// The hour of the day (local time) at which the scheduled day begins.
+const startHour = 6
+
+// StartOfDayOn gets the schedule start-of-day on a given date.
+// This is in terms of startHour.
+func StartOfDayOn(date time.Time) time.Time {
+	y, m, d := date.Date()
+	return time.Date(y, m, d, startHour, 0, 0, 0, time.Local)
+}
+
+// StartOffsetToHour takes a number of hours since the last day start (0-23) and gives the actual hour.
+// It returns an error if the hour is invalid.
+func StartOffsetToHour(hour int) (int, error) {
+	if 23 < hour || hour < 0 {
+		return 0, fmt.Errorf("StartOffsetToHour: hour %d not between 0 and 23", hour)
+	}
+	return (hour + startHour) % 24, nil
+}
+
+// HourToStartOffset takes an hour (0-23) and gives the number of hours elapsed since the last day start.
+// It returns an error if the hour is invalid.
+func HourToStartOffset(hour int) (int, error) {
+	if 23 < hour || hour < 0 {
+		return 0, fmt.Errorf("HourToStartOffset: hour %d not between 0 and 23", hour)
+	}
+	// Adding 24 to ensure we don't go negative.  Negative modulo is scary.
+	return ((hour + 24) - startHour) % 24, nil
+}
+
+//
+// Conversions from ISO year-week to more amenable formats.
+//
+
+// ParseIsoWeek parses an ISO weekday from year, week, and weekday strings.
+// It performs bounds checking.
+func ParseIsoWeek(year, week, weekday string) (int, int, time.Weekday, error) {
+	y, err := strconv.Atoi(year)
+	if err != nil {
+		return 0, 0, 0, err
```

Comment:
Named return values?

Reply:
I think that'd be good, if not just to explain what the 4 return values are